backend/app/services/article_vector_service.py

```diff
@@ -147,32 +127,15 @@ async def index_summarized_articles_to_vector_store(
         page: int = 0
         offset: int = page * page_size
 
-        articles: Sequence[Article] = (
+        articles: List[Article] = (
             await ArticleRepository.list_articles_with_summary(
                 limit=page_size, offset=offset
             )
         )
 
         while articles:
```

Comment:
```suggestion
        while len(articles) > 0:
```

I know it works in python as was, but it is general cleaner to be specific in your conditions. this would also work if articles is an object, True or other values, which may break other parts that follow after this in the app. for debugging, it's generally easier if the error is catched or breaks something closer to the origin, then a few steps down the line